### ISSUE TITLE
Changing "Try to recognize EPs and singles automatically" should trigger something?

### ISSUE THREAD

**Issue by darrell-k** [LMS-Community/slimserver#1211]
Triggering a rescan would obviously help, but could we get away with running some version of `Slim::Music::ReleaseTypes` without the progress update?

**Comment by michaelherger** [comment]
You could factor out all of the worker code into something like 

```perl
sub cleanupReleaseTypes {
   my ($class, $progress) = @_;

   $progress->update('yadda') if $progress;

   $do->('heavy work');

   $progress->final() if $progress;
```

Then call `$class->cleanupReleaseTypes($progress)` from `startScan()`, without the `$progress` object in cases where you don't want it. Does that make sense?

**Comment by darrell-k** [comment]
Or just condition the `$progress` stuff on `main::SCANNER` ?

**Comment by michaelherger** [comment]
...plus the `endImporter()`. Yes, that's another option.

**Comment by darrell-k** [comment]
Simpler for a simple person like me!

**Comment by michaelherger** [comment]
Running in the main process comes at the risk of halting playback: the database access is blocking. Larger collections might take a moment to process. You should add some of the following between the DB requests:

```perl
	main::idleStreams() unless main::SCANNER;
```

That would give the streams a bit of air to breath (or stream). Basically replace the progress with the above for non-scanner mode.

**Comment by darrell-k** [comment]
Actually, I think we're going to have to simply trigger a rescan, because when the pref is switched off, we have no way of distinguishing releases that the user actually tagged EP or SINGLE.

**Comment by michaelherger** [comment]
Closed by #1212
